fem/mdgridfunc.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_MDGRIDFUNC
+#define MFEM_MDGRIDFUNC
+
+#include "../config/config.hpp"
+
+#include "fem/gridfunc.hpp"
+#include "general/mdspan.hpp"
+
+namespace mfem
+{
+
+template<int N, class Layout = MDLayoutLeft<N>>
+class MDGridFunction : public MDSpan<GridFunction, N, Layout>
+{
+   using base_t = MDSpan<GridFunction, N, Layout>;
+   using base_t::Nd;
+   using base_t::Sd;
+   using GridFunction::data;
+
+public:
+
+   /**
+    * @brief MDGridFunction default constructor (recursion)
+    */
+   MDGridFunction(): base_t() { }
+
+   /**
+    * @brief MDGridFunction recursion constructor
+    * @param[in] fes Finite element space to use
+    * @param[in] args Rest of dimension indices
+    */
+   template <typename... Ts>
+   MDGridFunction(FiniteElementSpace *fes, Ts... args): MDGridFunction(args...)
+   {
```

Comment:
In the future, we'd like to be able to use different `FiniteElementSpaces` along the different dimensions in a Cartesian way.  For example, we'd use an L2 space for the physical space dimension, and another L2 1D space for the energy discretization mesh.  I guess we could have more constructors.  But something to think about.

Reply:
Correct, adding support for multiple finite element spaces should be relatively simple.